src/mlpack/methods/ann/layer/separable_convolution_impl.hpp

```diff
@@ -0,0 +1,495 @@
+              batchCount * inSize), weight.slice(outMapIdx), convOutput,
+              strideWidth, strideHeight);
+        }
+        else
+        {
+          ForwardConvolutionRule::Convolution(inputTemp.slice(inMap +
+              batchCount * inSize), weight.slice(outMapIdx), convOutput,
+              strideWidth, strideHeight);
+        }
+
+        outputTemp.slice(outMap) += convOutput;
+      }
+      outputTemp.slice(outMap) += bias(outMap % outSize);
+    }
+  }
+  outputWidth = outputTemp.n_rows;
+  outputHeight = outputTemp.n_cols;
+}
+
+template<
+    typename ForwardConvolutionRule,
+    typename BackwardConvolutionRule,
+    typename GradientConvolutionRule,
+    typename InputDataType,
+    typename OutputDataType
+>
+template<typename eT>
+void SeparableConvolution<
+    ForwardConvolutionRule,
+    BackwardConvolutionRule,
+    GradientConvolutionRule,
+    InputDataType,
+    OutputDataType
+>::Backward(
+    const arma::Mat<eT>& /* input */, arma::Mat<eT>& gy, arma::Mat<eT>& g)
+{
+  arma::cube mappedError(gy.memptr(), outputWidth, outputHeight,
+      outSize * batchSize, false, false);
+
+  g.set_size(inputTemp.n_rows * inputTemp.n_cols * inSize, batchSize);
+  gTemp = arma::Cube<eT>(g.memptr(), inputTemp.n_rows,
+      inputTemp.n_cols, inputTemp.n_slices, false, false);
+  gTemp.zeros();
+
+  for (size_t curGroup = 0; curGroup < numGroups; curGroup++)
+  {
+    for (size_t outMap = outSize * curGroup * batchSize / numGroups,
+        outMapIdx = outSize * curGroup / numGroups, batchCount = 0;
+        outMap < outSize * (curGroup + 1) * batchSize / numGroups; outMap++)
+    {
+      if (outMap != 0 && outMap % outSize == 0)
+      {
+        batchCount++;
+        outMapIdx = 0;
+      }
+
+      for (size_t inMap = inSize * curGroup / numGroups;
+          inMap < inSize * (curGroup + 1) / numGroups; inMap++, outMapIdx++)
+      {
+        arma::Mat<eT> output, rotatedFilter;
+        Rotate180(weight.slice(outMapIdx), rotatedFilter);
+        BackwardConvolutionRule::Convolution(mappedError.slice(outMap),
+        rotatedFilter, output, strideWidth, strideHeight);
+
+        if (padWLeft != 0 || padWRight != 0 || padHTop != 0 || padHBottom != 0)
+        {
+          gTemp.slice(inMap + batchCount * inSize) += output.submat(padWLeft,
+            padHTop, padWLeft + gTemp.n_rows - 1, padHTop + gTemp.n_cols - 1);
+        }
+        else
+        {
+          gTemp.slice(inMap + batchCount * inSize) += output;
+        }
+      }
+    }
+  }
+}
+
+template<
+    typename ForwardConvolutionRule,
+    typename BackwardConvolutionRule,
+    typename GradientConvolutionRule,
+    typename InputDataType,
+    typename OutputDataType
+>
+template<typename eT>
+void SeparableConvolution<
+    ForwardConvolutionRule,
+    BackwardConvolutionRule,
+    GradientConvolutionRule,
+    InputDataType,
+    OutputDataType
+>::Gradient(
+    const arma::Mat<eT>& /* input */,
+    arma::Mat<eT>& error,
```

Comment:
```suggestion
    const arma::Mat<eT>& error,
```

Maybe this should fix the issue.